tests/test_sycl_queue.py

```diff
@@ -1195,34 +1195,52 @@ def test_matrix_rank(device):
     assert_array_equal(expected, result)
 
 
+@pytest.mark.parametrize(
```

Comment:
What about QR tests in `tests/test_linalg.py`? I guess they needs to be updated and extended.